lib/matplotlib/image.py

```diff
@@ -1539,7 +1539,8 @@ def imsave(fname, arr, vmin=None, vmax=None, cmap=None, format=None,
             pil_shape = (rgba.shape[1], rgba.shape[0])
             image = Image.frombuffer(
                 "RGBA", pil_shape, rgba, "raw", "RGBA", 0, 1)
-            if format == "png" and metadata is not None:
+            if (format == "png"
+                    and metadata is not None and "pnginfo" not in pil_kwargs):
```

Comment:
Silently ignoring `metadata` if `pnginfo` is present does not feel right. Can we in that case, or are there subsets of metadata that may still be needed?

Reply:
print_png (used by `savefig("foo.png")` has had this behavior ever since pil_kwargs is supported (which is 3.1), and no one has complained about it so far (it probably hasn't been too much used yet in the wild, though); I'd rather keep both places in sync.
Also a verb is missing in your sentence ("Can we ???").

Reply:
Fixed the missing verb. I'd be happy to warn in both places. Raising would now be a breaking change.

I wouldn't expect too much complaint. It's not widely used, it will only happen if you provide both parameters. It might actually have affect users without them realizing it.

Reply:
sure, added a warning in both places.